internal/workload/podspec_updates.go

```diff
@@ -56,27 +59,42 @@ var l = logf.Log.WithName("internal.workload")
 // PodAnnotation returns the annotation (key, value) that should be added to
 // pods that are configured with this AuthProxyWorkload resource. This takes
 // into account whether the AuthProxyWorkload exists or was recently deleted.
-func PodAnnotation(r *cloudsqlapi.AuthProxyWorkload) (string, string) {
+// The defaultProxyImage is part of the annotation value.
+func PodAnnotation(r *cloudsqlapi.AuthProxyWorkload, defaultProxyImage string) (string, string) {
+	img := defaultProxyImage
+	if r.Spec.AuthProxyContainer != nil && r.Spec.AuthProxyContainer.Image != "" {
+		img = ""
+	}
 	k := fmt.Sprintf("%s/%s", cloudsqlapi.AnnotationPrefix, r.Name)
-	v := fmt.Sprintf("%d", r.Generation)
+	v := fmt.Sprintf("%d,%s", r.Generation, img)
 	// if r was deleted, use a different value
 	if !r.GetDeletionTimestamp().IsZero() {
-		v = fmt.Sprintf("%d-deleted-%s", r.Generation, r.GetDeletionTimestamp().Format(time.RFC3339))
+		v = fmt.Sprintf("%d-deleted-%s,%s", r.Generation, r.GetDeletionTimestamp().Format(time.RFC3339), img)
 	}
 
 	return k, v
 }
 
+// PodAnnotation returns the annotation (key, value) that should be added to
+// pods that are configured with this AuthProxyWorkload resource. This takes
+// into account whether the AuthProxyWorkload exists or was recently deleted.
+func (u *Updater) PodAnnotation(r *cloudsqlapi.AuthProxyWorkload) (string, string) {
```

Comment:
Why the exported method here to match the one above? Could we just have one of these exported?

Reply:
We need to be able to generate annotation keys and values for our tests, controlling the value of the default proxy image. `workload.PodAnnotation()` the exported package function is used in a bunch of test cases across the project. 

`func (u *Updater) PodAnnotation()` is used by the controller to generate the annotation key and value to use during reconcile. The `Updater` is responsible for knowing how to transform workloads, including value for default proxy image.